src/testRunner/compilerRunner.ts

```diff
@@ -480,13 +482,15 @@ class IsolatedDeclarationTest extends CompilerTestBase {
     protected isDiagnosticEquivalent: boolean;
     protected isOutputMapEquivalent: boolean;
 
-    static transformEnvironment(compilerEnvironment: CompilerTestEnvironment): CompilerTestEnvironment | undefined {
+    static transformEnvironment(compilerEnvironment: CompilerTestEnvironment, shouldNotExclude = !!compilerEnvironment.testCaseContent.settings.isolatedDeclarationDiffReason): CompilerTestEnvironment | undefined {
         const options = compilerEnvironment.compilerOptions;
-        // Exclude tests some tests
-        // - those explicitly not opting into isolatedDeclarations
-        // - those that do not usually emit output anyway
-        if (options.isolatedDeclarations === false || options.noEmit || options.noTypesAndSymbols || !options.declaration) {
-            return undefined;
+        if (!shouldNotExclude) {

```

Comment:
nit: Do we want to merge the nested if statments? It seems like you were trying to do that by covering the statements with brackets.

Reply:
Yeah. I did initially do that, but decided against it. The condition just looked complicated and had to follow. I removed the extra `()`